server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java

```diff
@@ -124,6 +127,17 @@ public Configuration(MailRepositoryUrl urlPrefix) {
         }
     }
 
+    @Override
+    public Publisher<User> usersWithVault() {
+        return Flux.fromStream(mailRepositoryStore.getUrls()
+            .filter(url -> url.hasPrefix(configuration.urlPrefix))
+            .map(MailRepositoryUrl::getPath)
+            .map(MailRepositoryPath::parts)
+            .peek(parts -> Preconditions.checkState(!parts.isEmpty()))
```

Comment:
could be filtered in the `isVault` function, don't you think?

Reply:
This line is never hit as hasPrefix ensures we have some parts